### ISSUE TITLE
WordNet sense keys mismatch

### ISSUE THREAD

**Issue by etienne-v** [nltk/nltk#2420]
Hi,

I'm having problems with retrieving the correct definitions from the nltk wordnet corpus for a given sense key.  Please see [this](https://github.com/getalp/UFSAC/issues/2) related issue from the UFSAC repo.

The current ntlk WordNet version is 3.0:
```
import nltk
nltk.download('wordnet')
from nltk.corpus import wordnet as wn

print(wn.get_version())
```
gives:
```
3.0
```

Looking at the official WordNet 3.0 database (http://wordnetcode.princeton.edu/3.0/WordNet-3.0.tar.gz), the key "drive%1:04:03::" has a synset ID of 00572489 (according to the `dict/index.sense` file), and the definition related to this synset ID is "hitting a golf ball off of a tee with a driver" (according to the `dict/data.noun` file).

The key associated with the above definition from the official WordNet online search is also "drive%1:04:03::" (along with "driving%1:04:03::"), even though the online search version is 3.1.

Firstly, using the key "drive%1:04:03::" to look up the definition in ntlk as follows
```
print(wn.synset_from_sense_key("drive%1:04:03::"))
print(wn.synset_from_sense_key("drive%1:04:03::").definition())
print(wn.synset_from_sense_key("drive%1:04:03::").offset())
```
gives:
```
Synset('campaign.n.02')
a series of actions advancing a principle or tending toward a particular end
798245
```
This definition and ID doesn't correspond to either the WordNet online search or the database.  However, querying nltk with the key "drive%1:04:06::" (replacing 03 with 06)
```
print(wn.synset_from_sense_key("drive%1:04:06::"))
print(wn.synset_from_sense_key("drive%1:04:06::").definition())
print(wn.synset_from_sense_key("drive%1:04:06::").offset())
```
gives the expected definition and ID:
```
Synset('drive.n.06')
hitting a golf ball off of a tee with a driver
572489
```
For this key, why doesn't ntlk give the definition as recorded in the database and the online search?

Also, when using the key "driving%1:04:03::" (referring to the second synonym for that sense), to obtain the synset with
```
print(wn.synset_from_sense_key("driving%1:04:03::"))
```
I get the following error:
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
~/anaconda3/lib/python3.7/site-packages/nltk/corpus/reader/wordnet.py in synset(self, name)
   1333         try:
-> 1334             offset = self._lemma_pos_offset_map[lemma][pos][synset_index]
   1335         except KeyError:

IndexError: list index out of range

During handling of the above exception, another exception occurred:

WordNetError                              Traceback (most recent call last)
<ipython-input-83-1778759d53e3> in <module>
----> 1 print(wn.synset_from_sense_key("driving%1:04:03::"))

~/anaconda3/lib/python3.7/site-packages/nltk/corpus/reader/wordnet.py in synset_from_sense_key(self, sense_key)
   1563 
   1564         synset_id = '.'.join([lemma, synset_types[int(ss_type)], lex_id])
-> 1565         return self.synset(synset_id)
   1566 
   1567     #############################################################

~/anaconda3/lib/python3.7/site-packages/nltk/corpus/reader/wordnet.py in synset(self, name)
   1343             else:
   1344                 tup = lemma, pos, n_senses, "senses"
-> 1345             raise WordNetError(message % tup)
   1346 
   1347         # load synset information from the appropriate file

WordNetError: lemma 'driving' with part of speech 'n' has only 2 senses
```

Any advice on the above issues will be much appreciated... thanks!

**Comment by alvations** [comment]
@etienne-v yes you're right about the error in the `synset_from_sense_key` function. 

First it calls a `parse_sense_key()` that breaks the sensekey into:

```
>>> from wn.reader import parse_sense_key
>>> parse_sense_key("drive%1:04:03::")
('drive', 'n', '03')
```

But when retrieving, it's not doing it right and retrieving the 3rd synset from the "drive" lemma, thus the `Synset('campaign.n.02')`

```
>>> wn.synset('drive.n.03')
Synset('campaign.n.02')
```

**Comment by etienne-v** [comment]
Thank you @alvations for the feedback.  Any suggestions on a workaround for this?

**Comment by tothniki** [comment]
Hi @etienne-v 
did you find any solution for this problem?

**Comment by ekaf** [comment]
Fixed by #2621
